docs/proposals/locking_notes.md

```diff
@@ -0,0 +1,193 @@
+
+# Create Locks Enhancement API proposal
+
+### Request
+
+```
+> POST https://lfs-server.com/locks
+> Accept: application/vnd.git-lfs+json
+> Content-Type: application/vnd.git-lfs+json
+> Authorization: Basic ...
+> {
+>  "path": "foo/bar.zip",
+>  "ref": {
+>    "name": "refs/heads/my-feature"
+>  },
+>  "notes": "Lock applied from Workspace A"
+> }
+```
+
+### Response
+
+* **Successful response**
+```
+< HTTP/1.1 201 Created
+< Content-Type: application/vnd.git-lfs+json
+< {
+<  "lock": {
+<    "id": "some-uuid",
+<    "path": "foo/bar.zip",
+<    "locked_at": "2022-05-17T15:49:06+00:00",
+<    "owner": {
+<      "name": "Jane Doe"
+<    },
+<    "notes": "Lock applied from Workspace A"
+<  }
+< }
+```
+
+# List Locks Enhancement API proposal
+
+### Request (with notes -- notes=true)
+
+```
+> GET https://lfs-server.com/locks?path=&id&cursor=limit&**notes=true**&refspec=
+> Accept: application/vnd.git-lfs+json
+> Authorization: Basic ... (if needed)
+```
+
+### Response
+
+* **Successful response**
+```
+< HTTP/1.1 200 Ok
+< Content-Type: application/vnd.git-lfs+json
+< {
+<  "locks": [
+<    {
+<      "id": "some-uuid",
+<      "path": "foo/bar.zip",
+<      "locked_at": "2022-05-17T15:49:06+00:00",
+<      "owner": {
+<        "name": "Jane Doe"
+<      },
+<      "notes": "Lock applied from Workspace A"
+<    }
+<  ],
+<  "next_cursor": "optional next ID"
+< }
+```
+
+### Request (with out notes)
+
+```
+> GET https://lfs-server.com/locks?path=&id&cursor=limit&refspec=
+> Accept: application/vnd.git-lfs+json
+> Authorization: Basic ... (if needed)
+```
+
+### Response
+
+* **Successful response**
+```
+< HTTP/1.1 200 Ok
+< Content-Type: application/vnd.git-lfs+json
+< {
+<  "locks": [
+<    {
+<      "id": "some-uuid",
+<      "path": "foo/bar.zip",
+<      "locked_at": "2022-05-17T15:49:06+00:00",
+<      "owner": {
+<        "name": "Jane Doe"
+<      }
+<    }
+<  ],
+<  "next_cursor": "optional next ID"
+< }
+```
+
+# List Locks for Verification Enhancement API proposal
+
+### Request (with notes)
+
+```
+> POST https://lfs-server.com/locks/verify
+> Accept: application/vnd.git-lfs+json
+> Content-Type: application/vnd.git-lfs+json
+> Authorization: Basic ...
+> {
+>  "cursor": "optional cursor",
+>  "limit": 100, // also optional
+>  "ref": {
+>    "name": "refs/heads/my-feature"
+>  },
+>  "notes" : true, // also optional
+> }
+```
+
+### Response
+
+* **Successful response**
+```
+< HTTP/1.1 200 Ok
+< Content-Type: application/vnd.git-lfs+json
+< {
+<  "ours": [
+<    {
+<      "id": "some-uuid",
+<      "path": "/path/to/file",
+<      "locked_at": "2016-05-17T15:49:06+00:00",
+<      "owner": {
+<        "name": "Jane Doe"
+<      },
+<      "notes": "Lock applied from Workspace A"
+<    }
+<  ],
+<  "theirs": [],
+<  "next_cursor": "optional next ID"
+< }
+```
+
+### Request (with out notes)
```

Comment:
Do we actually want this? Such feature (answering without notes after you added notes support) will actually require noticeable amounts of code. Both git-lfs and https://github.com/git-as-svn/git-lfs-java simply call JSON serializer on a Lock structure.

git-lfs: https://github.com/git-lfs/git-lfs/blob/v3.2.0/locking/locks.go#L200-L211
git-lfs-java: https://github.com/git-as-svn/git-lfs-java/blob/v0.19.0/gitlfs-common/src/main/kotlin/ru/bozaro/gitlfs/common/data/Lock.kt#L6-L30

And it is nontrivial to conditionally exclude a field from serialization.

The only reason to omit notes that I can come up with is network traffic.

Reply:
I think it's fine for the server to simply return any notes that are present, whether or not any were requested.  If no notes are requested on the client side, we'll just not emit the field, and then the server can either return them if appropriate or not.